src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -361,4 +361,126 @@ protected function inject_attribute( &$block ) {
 		}
 	}
 
+	/**
+	 * For compatibility reason, we need to wrap the Single Product template in a div with specific class.
+	 * For more details, see https://github.com/woocommerce/woocommerce-blocks/issues/8314.
+	 *
+	 * @param string $template_content Template Content.
+	 * @return string Wrapped template content inside a div.
+	 */
+	public static function wrap_single_product_template( $template_content ) {
+		$parsed_blocks   = parse_blocks( $template_content );
+		$splitted_blocks = self::split_blocks_by_type( $parsed_blocks );
+
+		$parsed_block_without_template_parts = array_filter(
+			$parsed_blocks,
+			function( $block ) {
+				return 'core/template-part' !== $block['blockName'] && ! empty( $block['blockName'] );
+			}
+		);
+
+		$single_product_template_blocks = array( 'woocommerce/breadcrumbs', 'woocommerce/product-gallery-image', 'woocommerce/product-add-to-cart', 'woocommerce/product-details', 'woocommerce/add-to-cart-form' );
```

Comment:
I would like to not hardcoded these slugs, but the name of the blocks isn't declared as constant :'(

Reply:
Because it is a recursive function, I avoided creating multiple instances of the same array.

Reply:
Aaah, makes sense!